### PR TITLE
Move subscription docs

### PR DESCRIPTION
**Pull request by usha-mandya**
Signed-off-by: Usha Mandya <usha.mandya@docker.com>

- Move Docker subscription docs from `/docker-hub/billing` to `/subscription/`
- Update Table of contents to point docs to the new location
- Add redirects from old locations
- Review and update existing references to point to /subscription/
- Delete `docker-hub/billing/` folder

### CODE REVIEW

**Comment by usha-mandya** [comment]
@thaJeztah @jerae-duffin PTAL

File: docker-hub/release-notes.md [hunk 1]
```diff
@@ -139,7 +139,7 @@ Each organization page now breaks down into these tabs:
 
 ### Enhancements
 
-* The [billing page](https://docs.docker.com/docker-hub/upgrade/) for personal accounts has been updated. You can access the page at its new URL: [https://hub.docker.com/billing/plan](https://hub.docker.com/billing/plan).
+* The [billing page](/subscription/upgrade.md) for personal accounts has been updated. You can access the page at its new URL: [https://hub.docker.com/billing/plan](https://hub.docker.com/billing/plan).
```
Comment (thaJeztah) [inline docker-hub/release-notes.md:142]:
```suggestion
* The [billing page](../subscription/upgrade.md) for personal accounts has been updated. You can access the page at its new URL: [https://hub.docker.com/billing/plan](https://hub.docker.com/billing/plan).
```

File: docker-hub/release-notes.md [hunk 2]
```diff
@@ -32,7 +32,7 @@ You can now specify any email address to receive billing-related emails for your
 
 To change the email address receiving billing-related emails, log into Docker Hub and navigate to the **Billing** tab of your organization. Select **Payment Methods** > **Contact Information**. Enter the new email address that you'd like to use in the **Email** field. Click **Update** for the changes to take effect.
 
-For details on how to update your billing information, see [Update billing information](/billing/index.md#update-billing-information).
+For details on how to update your billing information, see [Update billing information](/subscription/index.md#update-billing-information).
```
Comment (thaJeztah) [inline docker-hub/release-notes.md:35]:
```suggestion
For details on how to update your billing information, see [Update billing information](../subscription/index.md#update-billing-information).
```

**Review by thaJeztah — CHANGES_REQUESTED** [review]
left some comments on links, but LGTM otherwise

File: docker-hub/release-notes.md (hunk 1 above)
Comment (usha-mandya) [reply docker-hub/release-notes.md]: Thanks. Fixed

File: docker-hub/release-notes.md (hunk 2 above)
Comment (usha-mandya) [reply docker-hub/release-notes.md]: Fixed.

**Review by jerae-duffin — APPROVED** [review]
LGTM!

**Review by thaJeztah — APPROVED** [review]
LGTM

**Pull request merged**